---
title: "使用ggmsa可视化和探索多序列比对和相关数据"
author: "Lang Chau, Guangchuang Yu, Shuangbin Xu"
date: "`r format(Sys.time(), '%b %Y')`"
biblio-style: apalike
output:
  html_document:
    toc: true
    toc_float: true
    fig_caption: true
    code_folding: show
vignette: >
  %\VignetteIndexEntry{使用ggmsa可视化和探索多序列比对和相关数据}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>"
)
Biocpkg <- function (pkg){
    sprintf("[%s](http://bioconductor.org/packages/%s)", pkg, pkg)
}
CRANpkg <- function(pkg){
    cran <- "https://CRAN.R-project.org/package"
    fmt <- "[%s](%s=%s)"
    sprintf(fmt, pkg, cran, pkg)
}
hooks = knitr::knit_hooks$get()
hook_foldable = function(type) {
   force(type)
   function(x, options) {
     res = hooks[[type]](x, options)
     if (isFALSE(options[[paste0("fold.", type)]])) return(res)
     paste0(
       "<details><summary>", type, "</summary>\n\n",
       res,
       "\n\n</details>"
     )
   }
}
knitr::knit_hooks$set(
  output = hook_foldable("output"),
  plot = hook_foldable("plot")
)

#load package 
library(ggmsa)
library(ggplot2)
library(aplot)
library(phangorn)
library(ggimage)
library(ggtree)
library(readxl)
library(Biostrings)
library(ggnewscale)
library(RColorBrewer)
library(dplyr)
```


#  背景介绍

##  摘要 
在多序列比对(Multiple Sequence Alignment,MSA)的可视化中，普遍采用堆叠可视化的形
式，其以行表示每条序列，以列表示相同位置的不同残基字符。我们开发的R包ggmsa在
可视化层面扩展了这种堆叠形式的可视化方法，使得MSA图形可以与各类数据进行整合。
例如：允许将序列二级结构信息、基因座位点信息或表型数据等序列相关的数据集与MSA结
合，以此来探索序列-结构-功能三者之间的关系。并且，我们还开发和整合多种特别的
比对序列可视化方法，包核苷酸差异图、核苷酸相似性图 (鉴定序列重组信号)、
序列集束(Sequence Bundles)和序列标识(Sequence Logos)，
这允许用户从不同角度探索序列的特征。

##  生物学问题描述

###  issues 1 
使用单一的可视化范式很难捕获隐藏在排列中的令人满意的分子特征。

```{r, fig.height = 3, fig.width = 11, message=FALSE, warning=FALSE, dpi=300, echo=FALSE}
protein_sequences <- system.file("extdata", "sample.fasta", package = "ggmsa")
ggmsa(protein_sequences, 
      start = 221, 
      end = 280, 
      char_width = 0.5,
      seq_name = TRUE,
      border = NA)
```

堆叠图形在展示局部MSA时有很大优势，它以精确到单碱基/氨基酸水平的分辨率展示比对，
并通过染色让用户能快速识别比对的变化趋势。


```{r,  fig.height = 6, fig.width = 10, message=FALSE, warning=FALSE, dpi=300, echo=FALSE}
ggmsa("data/PlantMITERepeatMask.fa", 
      font = NULL, 
      start = 200,
      end = 300,
      seq_name = F,
      border = NA,
      color = "Chemistry_NT") + 
    coord_cartesian()
```


但是当比对序列比较大比较复杂时，会让人产生视觉上的混乱感。因为图形会包含密集的
字符和令人眼花缭乱的配色。

###  issues 2
现有工具缺乏同时探索MSA和相关数据的能力。

![](images/issus2.png#pic_center)

我们在探索MSA的序列特征时，往往试图深入序列的系统发育关系和结构功能关系。
因为它们时相互联系，相互影响。但是这在可视化层面就无法完成,
我们没有办法在绘图时把序列和其他数据联系起来。


#  使用ggmsa探索MSA和相关数据

##  ggmsa的技术路线
![](images/pipeline.png#pic_center)

ggmsa通过两种方案来解决上面的问题。
首先，单一的可视化方法无法得到令人满意的可视化方法，因此ggmsa开发并整合多种
通用的序列可视化方法。这为用户提供了更多可视化手段来从不同角度探索序列数据。

其次，ggmsa 拓展了普通的堆叠图形，使得外部数据可以对齐到MSA的行或者列上，实现
在可视化层面同时探索序列和相关数据。

##  不同的可视化方法

###  堆叠图形


```{r, echo=FALSE, out.width = "30%", fig.align='center'}
knitr::include_graphics("images/stacked_MSA.png")
```


```{r, fig.height = 3, fig.width = 11, message=FALSE, warning=FALSE, dpi=300}
protein_sequences <- system.file("extdata", "sample.fasta", package = "ggmsa")
readAAMultipleAlignment(protein_sequences) 
ggmsa(protein_sequences, 
      start = 221, 
      end = 280, 
      char_width = 0.5,
      seq_name = TRUE,
      border = NA)
```

这是最简单的序列可视化方法
应用的颜色，字体，输入数据类型

```{r}
available_colors()
available_fonts()
available_msa()
```


###  堆叠图形中的注释模块

```{r fig.height = 2.5, fig.width = 11, message=FALSE, warning=FALSE, dpi=300}
protein_sequences <- system.file("extdata", "sample.fasta", package = "ggmsa")
ggmsa(protein_sequences, start = 221, end = 280, char_width = 0.5, seq_name = TRUE) + geom_seqlogo() + geom_msaBar()
```

```{r  echo=FALSE, results='asis', warning=FALSE, message=FALSE}  

x <- "geom_seqlogo()\tgeometric layer\tautomatically generated sequence logos for a MSA\n
geom_GC()\tannotation module\tshows GC content with bubble chart\n
geom_seed()\tannotation module\thighlights seed region on miRNA sequences\n
geom_msaBar()\tannotation module\tshows sequences conservation by a bar chart\n
geom_helix()\tannotation module\tdepicts RNA secondary structure as arc diagrams(need extra data)\n
 "
require(dplyr)
xx <- strsplit(x, "\n\n")[[1]]
y <- strsplit(xx, "\t") %>% do.call("rbind", .)
y <- as.data.frame(y, stringsAsFactors = FALSE)

colnames(y) <- c("Annotation modules", "Type", "Description")

require(kableExtra)
knitr::kable(y, align = "l", booktabs = TRUE, escape = TRUE) %>% 
    kable_styling(latex_options = c("striped", "hold_position", "scale_down"))
  
```


不同的布局

```{r facet, fig.height = 6, fig.width = 10, message=FALSE, warning=FALSE, fig.cap="\\label{fig:facet} The long sequence was broken down and displayed in several lines."}
 # 4 fields
 ggmsa(protein_sequences, start = 0, end = 400, font = NULL, color = "Chemistry_AA") + 
    facet_msa(field = 100)
```


###  Sequence bundle 

```{r, echo=FALSE, out.width = "30%", fig.align='center'}
knitr::include_graphics("images/seq_bundle.png")
```

```{r echo=FALSE, fig.height=5.5, fig.width=11, message=FALSE}
negative <-  system.file("extdata", "Gram-negative_AKL.fasta", 
                         package = "ggmsa")
positive <-  system.file("extdata", "Gram-positive_AKL.fasta", 
                         package = "ggmsa")

pos <- data.frame(x= c(4, 7, 9, 24, 27, 29,
                       4, 7, 24, 27), 
                  y = c(c(21, 11, 20, 17, 12, 18) +.3,
                        c(13, 13, 13, 13) +.5
                        ), 
                  label = c("H", "S", "R", "D", "T", "E",
                            "C", "C", "C", "C"),
                  color = c(rep("#ff4700",6), 
                            rep("#0443d0",4)))

ggSeqBundle(list(negative, positive), 
            alpha = 0.1, 
            bundle_color = c("#FC8D62","#8DA0CB"))+ #RColorBrewer: Set2:2-3
    geom_text(data = pos, 
              mapping = aes(x, y, 
                            label = label, 
                            color = I(color)),
              inherit.aes = FALSE, 
              size = 4)

```

###  Sequence logo

```{r, echo=FALSE, out.width = "30%", fig.align='center'}
knitr::include_graphics("images/seq_logo.png")
```


```{r fig.height = 3, fig.width = 11, warning = FALSE, message = FALSE}
 ggmsa(protein_sequences, 221, 280, seq_name = TRUE, char_width = 0.5, font = NULL) +
    geom_seqlogo(color = "Chemistry_AA") 
```

```{r fig.height = 2.5, fig.width = 8, warning = FALSE, message = FALSE}
 seqlogo(protein_sequences, 221, 280)
```


###  Sequence recombination

```{r, echo=FALSE, out.width = "30%",out.height='50%', fig.align='center'}
knitr::include_graphics("images/recombination.png")
```


```{r  fig.height = 11, fig.width = 10, warning=FALSE, message=FALSE}
fas <- list.files(system.file("extdata", "GVariation", package="ggmsa"),
                  pattern="fas", full.names=TRUE)
xx <- lapply(fas, seqdiff)
plts <- lapply(xx, plot, width = 100)
fas[4] <- system.file("extdata/GVariation/sample_alignment.fa", 
                    package="ggmsa")# + theme(legend.position = "bottom")
plts[[4]] <- simplot(fas[4], 'CF_YL21')
plot_list(gglist=plts, ncol=1, tag_levels = list(c("A",' ',"B", ' ',"C", ' ', "D")))
```


##  拓展Stacked MSA

![](images/Fig2.png)

###  对齐到行

```{r  fig.height = 4, fig.width = 15, warning=FALSE}
#load MSA and genes locus
tp53_sequences <-  system.file("extdata", "tp53.fa", package = "ggmsa")
tp53_genes <- system.file("extdata", "TP53_genes.xlsx", package = "ggmsa")

#generating phylogenetic tree according to MSA
dat <- read.aa(tp53_sequences, format = "fasta") %>% phyDat(type = "AA", levels = NULL)
tree <- dist.ml(dat, model = "JTT") %>% bionj()
dd <- ggimage::phylopic_uid(tree$tip.label)

#visualizing phylogenetic tree by ggtree
p_tp53 <- ggtree(tree, branch.length = 'none') %<+% dd +
  geom_tiplab(aes(image=uid), geom = "phylopic", offset =1.9) +
  geom_tiplab(aes(label=label)) +
  geom_treescale(x = 0,y = -1)

#deal with MSA data
data_53 <- readAAMultipleAlignment(tp53_sequences) %>% tidy_msa()

#gene maps
TP53_arrow <- read_xlsx(tp53_genes)
TP53_arrow$direction <- 1
TP53_arrow[TP53_arrow$strand == "reverse","direction"] <- -1

#color
mapping = aes(xmin = start, xmax = end, fill = gene, forward = direction)
my_pal <- colorRampPalette(rev(brewer.pal(n = 10, name = "Set3")))

#tree + gene maps + msa
p6a <- p_tp53  + xlim_tree(4) +
  geom_facet(geom = geom_msa, data = data_53,
             panel = 'Multiple Sequence Alignment of the TP53 Protein', font = NULL,
             border = NA) +
  new_scale_fill() +
  scale_fill_manual(values = my_pal(10)) +
  geom_facet(geom = geom_motif,
             mapping = mapping, data = TP53_arrow,
             panel = 'Genome_Locus',  on = 'TP53',
             arrowhead_height = unit(3, "mm"),
             arrowhead_width = unit(1, "mm")) +
  theme(strip.background=element_blank(),
        strip.text = element_text(size = 13))
p6A <- facet_widths(p6a, c(Tree = 0.35, Genome_Locus = 0.3))
p6A
```

###  对齐到列
```{r  fig.height = 14, fig.width = 10, warning=FALSE}
RNA7S  <- "data/3JAJ-2D-dotbracket.txt"
RNAP54 <- "data/4UJE-2D-dotbracket.txt"

RF03120_msa<- system.file("extdata", "Rfam", "RF03120.fasta", package = "ggmsa")
RF03120_ss <- system.file("extdata", "Rfam", "RF03120_SS.txt", package = "ggmsa")

known <- readSSfile(RNA7S, type = "Vienna" )
transat <- readSSfile(RNAP54 , type = "Vienna")


RF_arc <- readSSfile(RF03120_ss, type = "Vienna" )

p7A <- ggmsa(RF03120_msa, 
             font = NULL, 
             color = "Chemistry_NT", 
             seq_name = F, 
             show.legend = F, 
             border = NA) +
        geom_helix(helix_data = RF_arc) + 
        theme(axis.text.y = element_blank())

p7B <- ggmsa("data/5SRNA.fa",
             font = NULL,
             color = "Chemistry_NT",
             seq_name = T,
             show.legend = T,
             border = NA) +
  geom_helix(helix_data = list(known = known, 
                               predicted = transat),
             overlap = F)

p7 <- plot_list(gglist = list(p7A, p7B), 
                ncol = 1,
                heights = c(0.15), 
                tag_levels = 'A') 
p7
```


#  总结





