<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>使用ggmsa可视化和探索多序列比对和相关数据 • ggmsaABiocWorkshop2021</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="使用ggmsa可视化和探索多序列比对和相关数据">
<meta property="og:description" content="ggmsaABiocWorkshop2021">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ggmsaABiocWorkshop2021</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/workflow_ggmsa_Chinese.html">使用ggmsa可视化和探索多序列比对和相关数据</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/nyzhoulang/ggmsaBioWorkshop2021">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="workflow_ggmsa_Chinese_files/header-attrs-2.11/header-attrs.js"></script><script src="workflow_ggmsa_Chinese_files/kePrint-0.0.1/kePrint.js"></script><link href="workflow_ggmsa_Chinese_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>使用ggmsa可视化和探索多序列比对和相关数据</h1>
                        <h4 class="author">Lang Chau, Guangchuang Yu, Shuangbin Xu and Huina Huang</h4>
            
            <h4 class="date">Nov 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/nyzhoulang/ggmsaBioWorkshop2021/blob/master/vignettes/workflow_ggmsa_Chinese.Rmd"><code>vignettes/workflow_ggmsa_Chinese.Rmd</code></a></small>
      <div class="hidden name"><code>workflow_ggmsa_Chinese.Rmd</code></div>

    </div>

    
    
<div id="背景介绍" class="section level1">
<h1 class="hasAnchor">
<a href="#%E8%83%8C%E6%99%AF%E4%BB%8B%E7%BB%8D" class="anchor"></a>背景介绍</h1>
<div id="摘要" class="section level2">
<h2 class="hasAnchor">
<a href="#%E6%91%98%E8%A6%81" class="anchor"></a>摘要</h2>
<p>多序列比对(Multiple Sequence Alignment,MSA)常被用于探索比对序列中的保守区域和变 异区域。而堆叠图形(Stacked Graphs)是在探索过程中最常用的可视化方式。其以行表示 每条序列，以列表示相同位置的不同残基字符。我们开发的R包<code>ggmsa</code>扩展了这种堆叠形 式的可视化方法，并添加了更多可视化方法使用户能更好的观测到MSA的保守区域和变异区 域。另外，<code>ggmsa</code>结合<code>ggtree</code>和<code>ggtreeExtra</code>，可以在大分子层面，观察MSA与表型的 联系。总之，<code>ggmsa</code>可以为用户提供的更多的可视化手段，允许用户从不同角度探索序列 特征。</p>
</div>
<div id="ggmsa的应用场景" class="section level2">
<h2 class="hasAnchor">
<a href="#ggmsa%E7%9A%84%E5%BA%94%E7%94%A8%E5%9C%BA%E6%99%AF" class="anchor"></a>ggmsa的应用场景</h2>
<div id="msa保守和变异位点的可视化" class="section level3">
<h3 class="hasAnchor">
<a href="#msa%E4%BF%9D%E5%AE%88%E5%92%8C%E5%8F%98%E5%BC%82%E4%BD%8D%E7%82%B9%E7%9A%84%E5%8F%AF%E8%A7%86%E5%8C%96" class="anchor"></a>MSA保守和变异位点的可视化</h3>
<p>1.MSA中的保守区域:</p>
<ul>
<li>Sequence Bundles</li>
<li>Sequence Logos</li>
<li>Stacked MSA <code>+</code> Annotations</li>
</ul>
<p>2.序列中的特殊变异区域</p>
<ul>
<li>Sequence Recombination</li>
</ul>
<!-- 例如：最常用的堆叠模式方法也不能适用于所有情况： --><!-- ```{r, fig.height = 3, fig.width = 11, message=FALSE, warning=FALSE, dpi=300, echo=FALSE} --><!-- protein_sequences <- system.file("extdata", "sample.fasta", package = "ggmsa") --><!-- ggmsa(protein_sequences,  --><!--       start = 221,  --><!--       end = 280,  --><!--       char_width = 0.5, --><!--       seq_name = TRUE, --><!--       border = NA) --><!-- ``` --><!-- 堆叠图形在展示局部MSA时有很大优势，它以精确到单核苷酸/氨基酸水平的分辨率展示比对， --><!-- 并通过染色让用户能快速识别MSA的变化趋势。 --><!-- ```{r,  fig.height = 6, fig.width = 10, message=FALSE, warning=FALSE, dpi=300, echo=FALSE} --><!-- ggmsa("data/PlantMITERepeatMask.fa",  --><!--       font = NULL,  --><!--       start = 200, --><!--       end = 300, --><!--       seq_name = F, --><!--       border = NA, --><!--       color = "Chemistry_NT") +  --><!--     coord_cartesian() --><!-- ``` --><!-- 但是当比对序列比较大并且比较复杂时，图形会包含密集的字符和令人眼花缭乱的配色。 --><!-- 会让人产生视觉上的混乱感。 -->
</div>
<div id="探索msa和大分子功能的相关性分子水平" class="section level3">
<h3 class="hasAnchor">
<a href="#%E6%8E%A2%E7%B4%A2msa%E5%92%8C%E5%A4%A7%E5%88%86%E5%AD%90%E5%8A%9F%E8%83%BD%E7%9A%84%E7%9B%B8%E5%85%B3%E6%80%A7%E5%88%86%E5%AD%90%E6%B0%B4%E5%B9%B3" class="anchor"></a>探索MSA和大分子功能的相关性（分子水平）</h3>
<ul>
<li>Tree + MSA + Phenotypes</li>
</ul>
<p>序列，结构，功能是相互联系，相互影响的，基因组的变异可能可以解释表型的变异。 MSA可视化使基因组位点变异信息在视觉上被观测到；基因位置变异信息可以以箭头表示， 其他表型信息也可以用各种可视化方法表示出来。于是，从在可视化层面的将他们结合起来 是完全可行的。这我们可以在可视化层面观察到的序列变异与特殊表型的关联。 <img src="images/issus2.png"></p>
</div>
</div>
<div id="ggmsa的技术路线" class="section level2">
<h2 class="hasAnchor">
<a href="#ggmsa%E7%9A%84%E6%8A%80%E6%9C%AF%E8%B7%AF%E7%BA%BF" class="anchor"></a>ggmsa的技术路线</h2>
<p><img src="images/pipeline.png"></p>
<p>ggmsa中的众多可视化方法主要执行两大功能：</p>
<ul>
<li>在单核苷酸/氨基酸水平更好的展示MSA的保守区域和变异区域；</li>
<li>在分子水平的观察MSA和相关数据的联系；</li>
</ul>
</div>
</div>
<div id="示例演示" class="section level1">
<h1 class="hasAnchor">
<a href="#%E7%A4%BA%E4%BE%8B%E6%BC%94%E7%A4%BA" class="anchor"></a>示例演示</h1>
<div id="使用不同的可视化方法探索msa的保守性" class="section level2">
<h2 class="hasAnchor">
<a href="#%E4%BD%BF%E7%94%A8%E4%B8%8D%E5%90%8C%E7%9A%84%E5%8F%AF%E8%A7%86%E5%8C%96%E6%96%B9%E6%B3%95%E6%8E%A2%E7%B4%A2msa%E7%9A%84%E4%BF%9D%E5%AE%88%E6%80%A7" class="anchor"></a>使用不同的可视化方法探索MSA的保守性</h2>
<div id="堆叠图形" class="section level3">
<h3 class="hasAnchor">
<a href="#%E5%A0%86%E5%8F%A0%E5%9B%BE%E5%BD%A2" class="anchor"></a>堆叠图形</h3>
<p>堆叠图形以行表示每条序列，以列表示相同位置的不同残基字符，并根据不同的残基属性， 为字符渲染不同的颜色。通过着色可以快速识别变化趋势。</p>
<details><summary>
plot
</summary><p><img src="images/stacked_MSA.png" width="30%" style="display: block; margin: auto;"></p>
</details><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">protein_sequences</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sample.fasta"</span>, package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>
<span class="fu">readAAMultipleAlignment</span><span class="op">(</span><span class="va">protein_sequences</span><span class="op">)</span> <span class="co">#苯丙氨酸羟化酶蛋白</span></code></pre></div>
<details><summary>
output
</summary><pre><code>#&gt; AAMultipleAlignment with 9 rows and 456 columns
#&gt;      aln                                                    names               
#&gt; [1] MAAVVLENGVLSRKLSDFGQETSYIE...QLKILADSINSEVGILCNALQKIKS PH4H_Rattus_norve...
#&gt; [2] MAAVVLENGVLSRKLSDFGQETSYIE...QLKILADSINSEVGILCHALQKIKS PH4H_Mus_musculus
#&gt; [3] MSTAVLENPGLGRKLSDFGQETSYIE...QLKILADSINSEIGILCSALQKIK- PH4H_Homo_sapiens
#&gt; [4] MSALVLESRALGRKLSDFGQETSYIE...QLKILADSISSEVEILCSALQKLK- PH4H_Bos_taurus
#&gt; [5] --------------------------...LNAGDRQGWADTEDV---------- PH4H_Chromobacter...
#&gt; [6] --------------------------...LNAGTREGWADTADI---------- PH4H_Ralstonia_so...
#&gt; [7] --------------------------...LTRGT-QAYATAGGRLAGAAAG--- PH4H_Caulobacter_...
#&gt; [8] --------------------------...------------------------- PH4H_Pseudomonas_...
#&gt; [9] --------------------------...------------------------- PH4H_Rhizobium_loti</code></pre>
</details><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggmsa</span><span class="op">(</span><span class="va">protein_sequences</span>, 
      start <span class="op">=</span> <span class="fl">221</span>, 
      end <span class="op">=</span> <span class="fl">280</span>, 
      char_width <span class="op">=</span> <span class="fl">0.5</span>,
      seq_name <span class="op">=</span> <span class="cn">TRUE</span>,
      border <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></code></pre></div>
<details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/unnamed-chunk-3-1.png" width="3300"></p>
</details><p><code>ggmsa</code>函数作为基础函数，接收的MSA数据，控制MSA的可视化范围，配色，字体等参数。 参数选择可以用<code>available</code>系列函数来查看。</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">available_colors</span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; 1.color schemes for nucleotide sequences currently available:</span>
<span class="co">#&gt; Chemistry_NT Shapely_NT Taylor_NT Zappo_NT</span>
<span class="co">#&gt; 2.color schemes for AA sequences currently available:</span>
<span class="co">#&gt; ClustalChemistry_AA Shapely_AA Zappo_AA Taylor_AA LETTER CN6 Hydrophobicity</span>
<span class="fu">available_fonts</span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; font families currently available:</span>
<span class="co">#&gt; helvetical mono TimesNewRoman DroidSansMono</span>
<span class="fu">available_msa</span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; 1.files currently available:</span>
<span class="co">#&gt; .fasta</span>
<span class="co">#&gt; 2.XStringSet objects from 'Biostrings' package:</span>
<span class="co">#&gt; DNAStringSet RNAStringSet AAStringSet BStringSet DNAMultipleAlignment RNAMultipleAlignment AAMultipleAlignment</span>
<span class="co">#&gt; 3.bin objects:</span>
<span class="co">#&gt; DNAbin AAbin</span></code></pre></div>
</div>
<div id="堆叠图形中的注释模块" class="section level3">
<h3 class="hasAnchor">
<a href="#%E5%A0%86%E5%8F%A0%E5%9B%BE%E5%BD%A2%E4%B8%AD%E7%9A%84%E6%B3%A8%E9%87%8A%E6%A8%A1%E5%9D%97" class="anchor"></a>堆叠图形中的注释模块</h3>
<p>注释系统也是ggmsa中的一大亮点。类似于<code>ggplot2</code>中图形语法，我们使用 <code>ggmsa()</code> + <code>geom_*()</code>的模式添加注释。不同的<code>geom</code>图层对应不同的注释方法。</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggmsa</span><span class="op">(</span><span class="va">protein_sequences</span>, 
      start <span class="op">=</span> <span class="fl">221</span>, 
      end <span class="op">=</span> <span class="fl">280</span>, 
      char_width <span class="op">=</span> <span class="fl">0.5</span>, 
      seq_name <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">geom_seqlogo</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> 
  <span class="fu">geom_msaBar</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/unnamed-chunk-5-1.png" width="3300"></p>
</details>
并且不同注释图层是可以叠加使用的：
<table class="table table">
<thead><tr>
<th style="text-align:left;">
Annotation modules
</th>
<th style="text-align:left;">
Type
</th>
<th style="text-align:left;">
Description
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
geom_seqlogo()
</td>
<td style="text-align:left;">
geometric layer
</td>
<td style="text-align:left;">
automatically generated sequence logos for a MSA
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_seed()
</td>
<td style="text-align:left;">
annotation module
</td>
<td style="text-align:left;">
highlights seed region on miRNA sequences
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_msaBar()
</td>
<td style="text-align:left;">
annotation module
</td>
<td style="text-align:left;">
shows sequences conservation by a bar chart
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_helix()
</td>
<td style="text-align:left;">
annotation module
</td>
<td style="text-align:left;">
depicts RNA secondary structure as arc diagrams(need extra data)
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
</tbody>
</table>
<p>注释模块还可以接收外部数据。RNA的二级结构可以以弧线图的形式添加到MSA。 弧线连接的两个MSA位置表示在结构上是配对的。</p>
<p>使用<code>ggmsa() + geom_helix()</code>的模式可以将二级结构数据整合到MSA图形中。</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">RNA7S</span>  <span class="op">&lt;-</span> <span class="st">"data/3JAJ-2D-dotbracket.txt"</span>
<span class="va">RNAP54</span> <span class="op">&lt;-</span> <span class="st">"data/4UJE-2D-dotbracket.txt"</span>

<span class="va">RF03120_msa</span><span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"Rfam"</span>, <span class="st">"RF03120.fasta"</span>, package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>
<span class="va">RF03120_ss</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"Rfam"</span>, <span class="st">"RF03120_SS.txt"</span>, package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>

<span class="va">known</span> <span class="op">&lt;-</span> <span class="fu">readSSfile</span><span class="op">(</span><span class="va">RNA7S</span>, type <span class="op">=</span> <span class="st">"Vienna"</span> <span class="op">)</span>
<span class="va">transat</span> <span class="op">&lt;-</span> <span class="fu">readSSfile</span><span class="op">(</span><span class="va">RNAP54</span> , type <span class="op">=</span> <span class="st">"Vienna"</span><span class="op">)</span>


<span class="va">RF_arc</span> <span class="op">&lt;-</span> <span class="fu">readSSfile</span><span class="op">(</span><span class="va">RF03120_ss</span>, type <span class="op">=</span> <span class="st">"Vienna"</span> <span class="op">)</span>

<span class="va">p7A</span> <span class="op">&lt;-</span> <span class="fu">ggmsa</span><span class="op">(</span><span class="va">RF03120_msa</span>, 
             font <span class="op">=</span> <span class="cn">NULL</span>, 
             color <span class="op">=</span> <span class="st">"Chemistry_NT"</span>, 
             seq_name <span class="op">=</span> <span class="cn">F</span>, 
             show.legend <span class="op">=</span> <span class="cn">F</span>, 
             border <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op">+</span>
        <span class="fu">geom_helix</span><span class="op">(</span>helix_data <span class="op">=</span> <span class="va">RF_arc</span><span class="op">)</span> <span class="op">+</span> 
        <span class="fu">theme</span><span class="op">(</span>axis.text.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span>

<span class="va">p7B</span> <span class="op">&lt;-</span> <span class="fu">ggmsa</span><span class="op">(</span><span class="st">"data/5SRNA.fa"</span>,
             font <span class="op">=</span> <span class="cn">NULL</span>,
             color <span class="op">=</span> <span class="st">"Chemistry_NT"</span>,
             seq_name <span class="op">=</span> <span class="cn">T</span>,
             show.legend <span class="op">=</span> <span class="cn">T</span>,
             border <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_helix</span><span class="op">(</span>helix_data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>known <span class="op">=</span> <span class="va">known</span>, 
                               predicted <span class="op">=</span> <span class="va">transat</span><span class="op">)</span>,
             overlap <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>

<span class="va">p7</span> <span class="op">&lt;-</span> <span class="fu">plot_list</span><span class="op">(</span>gglist <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">p7A</span>, <span class="va">p7B</span><span class="op">)</span>, 
                ncol <span class="op">=</span> <span class="fl">1</span>,
                heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.15</span><span class="op">)</span>, 
                tag_levels <span class="op">=</span> <span class="st">'A'</span><span class="op">)</span> 
<span class="va">p7</span></code></pre></div>
<details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/unnamed-chunk-7-1.png" width="960"></p>
</details>
</div>
<div id="堆叠图形中的布局" class="section level3">
<h3 class="hasAnchor">
<a href="#%E5%A0%86%E5%8F%A0%E5%9B%BE%E5%BD%A2%E4%B8%AD%E7%9A%84%E5%B8%83%E5%B1%80" class="anchor"></a>堆叠图形中的布局</h3>
<p>除了不同的注释图层，我们还为堆叠图形设计不同的布局</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="co"># 4 fields</span>
 <span class="fu">ggmsa</span><span class="op">(</span><span class="va">protein_sequences</span>, start <span class="op">=</span> <span class="fl">0</span>, end <span class="op">=</span> <span class="fl">400</span>, font <span class="op">=</span> <span class="cn">NULL</span>, color <span class="op">=</span> <span class="st">"Chemistry_AA"</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu">facet_msa</span><span class="op">(</span>field <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></code></pre></div>
<details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/facet-1.png" width="960"></p>
</details>
</div>
<div id="sequence-bundle" class="section level3">
<h3 class="hasAnchor">
<a href="#sequence-bundle" class="anchor"></a>Sequence bundle</h3>
Sequence bundles可视化方法在2013年BioVis设计竞赛中被引入，以探索序列之间的残基 关系。MSA的序列被描绘成不间断的、半不透明的线。每条线代表一个序列。 因此整个MSA以序列束的形式被绘制，在y轴上的序列字母表示不同的残基符号并根据不同 的生化特性排序。x轴上代表序列位置。
<details><summary>
plot
</summary><p><img src="images/seq_bundle.png" width="30%" style="display: block; margin: auto;"></p>
</details><p>ggmsa中用<code>ggSeqBundle</code>函数来把MSA可视化为Sequence bundles.最简单的方法是 <code>ggSeqBundle(msa_data)</code>。</p>
<p>下列数据包括革兰氏阳性菌和革兰氏阴性菌的腺苷酸激酶盖(Adenylate Kinase Lid, AKL) 结构域。每组各100条序列。 阳性菌在形成半胱氨酸(Cystine)四联体；阴性菌形成氢键网络， (His4, Ser7, Arg9, Asp24, Thr27, Glu29)</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">negative</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"Gram-negative_AKL.fasta"</span>, 
                         package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>
<span class="va">positive</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"Gram-positive_AKL.fasta"</span>, 
                         package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>

<span class="va">pos</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/data.frame.html">data.frame</a></span><span class="op">(</span>x<span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">4</span>, <span class="fl">7</span>, <span class="fl">9</span>, <span class="fl">24</span>, <span class="fl">27</span>, <span class="fl">29</span>,
                       <span class="fl">4</span>, <span class="fl">7</span>, <span class="fl">24</span>, <span class="fl">27</span><span class="op">)</span>, 
                  y <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">21</span>, <span class="fl">11</span>, <span class="fl">20</span>, <span class="fl">17</span>, <span class="fl">12</span>, <span class="fl">18</span><span class="op">)</span> <span class="op">+</span><span class="fl">.3</span>,
                        <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">13</span>, <span class="fl">13</span>, <span class="fl">13</span>, <span class="fl">13</span><span class="op">)</span> <span class="op">+</span><span class="fl">.5</span>
                        <span class="op">)</span>, 
                  label <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"H"</span>, <span class="st">"S"</span>, <span class="st">"R"</span>, <span class="st">"D"</span>, <span class="st">"T"</span>, <span class="st">"E"</span>,
                            <span class="st">"C"</span>, <span class="st">"C"</span>, <span class="st">"C"</span>, <span class="st">"C"</span><span class="op">)</span>,
                  color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"#ff4700"</span>,<span class="fl">6</span><span class="op">)</span>, 
                            <span class="fu"><a href="https://rdrr.io/r/base/rep.html">rep</a></span><span class="op">(</span><span class="st">"#0443d0"</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="fu">ggSeqBundle</span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">negative</span>, <span class="va">positive</span><span class="op">)</span>, 
            alpha <span class="op">=</span> <span class="fl">0.1</span>, 
            bundle_color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"#FC8D62"</span>,<span class="st">"#8DA0CB"</span><span class="op">)</span><span class="op">)</span><span class="op">+</span> <span class="co">#RColorBrewer: Set2:2-3</span>
    <span class="fu">geom_text</span><span class="op">(</span>data <span class="op">=</span> <span class="va">pos</span>, 
              mapping <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span><span class="va">x</span>, <span class="va">y</span>, 
                            label <span class="op">=</span> <span class="va">label</span>, 
                            color <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/AsIs.html">I</a></span><span class="op">(</span><span class="va">color</span><span class="op">)</span><span class="op">)</span>,
              inherit.aes <span class="op">=</span> <span class="cn">FALSE</span>, 
              size <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></code></pre></div>
<details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/unnamed-chunk-9-1.png" width="1056"></p>
</details><p>(Date from BioVis2013 and repeated example from <a href="https://www.science-practice.com/projects/sequence-bundles/gallery/">Science Practice</a>)</p>
</div>
<div id="sequence-logo" class="section level3">
<h3 class="hasAnchor">
<a href="#sequence-logo" class="anchor"></a>Sequence logo</h3>
<p>Sequence logo可视化方法计算了MSA中每列字符的的频率，并把每列的字符根据频率缩放， 进行堆叠。</p>
<details><summary>
plot
</summary><p><img src="images/seq_logo.png" width="30%" style="display: block; margin: auto;"></p>
</details><p>ggmsa中的Sequence logo可以作为注释图层添<code>geom_seqlogo</code>添加到堆叠图形中。</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="fu">ggmsa</span><span class="op">(</span><span class="va">protein_sequences</span>, <span class="fl">221</span>, <span class="fl">280</span>, seq_name <span class="op">=</span> <span class="cn">TRUE</span>, char_width <span class="op">=</span> <span class="fl">0.5</span>, font <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">geom_seqlogo</span><span class="op">(</span>color <span class="op">=</span> <span class="st">"Chemistry_AA"</span><span class="op">)</span> </code></pre></div>
<details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/unnamed-chunk-11-1.png" width="1056"></p>
</details><p>也可以用<code>seqlogo</code>函数进行单独的渲染。使用方法与<code>ggmsa</code>相似，可是通过参数控制位置， 配色和字体。<code>patchwork</code>包可以把sequqnce logos 和 sequence bundles组合起来。</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="fu">seqlogo</span><span class="op">(</span><span class="va">protein_sequences</span>, <span class="fl">221</span>, <span class="fl">280</span><span class="op">)</span></code></pre></div>
<details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/unnamed-chunk-12-1.png" width="768"></p>
</details>
</div>
</div>
<div id="序列的中特殊的变异" class="section level2">
<h2 class="hasAnchor">
<a href="#%E5%BA%8F%E5%88%97%E7%9A%84%E4%B8%AD%E7%89%B9%E6%AE%8A%E7%9A%84%E5%8F%98%E5%BC%82" class="anchor"></a>序列的中特殊的变异</h2>
除了通过可视化方法探索MSA的保守性,<code>ggmsa</code>还设计专门的可视化方法来观测序列中特殊 的变异位点。为鉴定序列中的潜在的重组信号提供线索。
<details><summary>
plot
</summary><p><img src="images/recombination.png" width="30%" height="50%" style="display: block; margin: auto;"></p>
</details><p>选择一个潜在序列(query)，<code>seqdiff</code>可以计算query和其他序列在每个位置上是否存在 核苷酸差异。并把固定区域的差异数目以柱状图的形式展示(<code>plot(seqdiff)</code>)。</p>
<p>同理，用<code>simplot</code>函数计算序列间的相似性。例如：<code>CF_YL21</code> vs <code>Mont,Oz and Wilg5</code></p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GVariation"</span>, package<span class="op">=</span><span class="st">"ggmsa"</span><span class="op">)</span>,
                  pattern<span class="op">=</span><span class="st">"fas"</span>, full.names<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">xx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">fas</span>, <span class="va">seqdiff</span><span class="op">)</span>
<span class="va">plts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">xx</span>, <span class="va">plot</span>, width <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>
<span class="va">fas</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/GVariation/sample_alignment.fa"</span>, 
                    package<span class="op">=</span><span class="st">"ggmsa"</span><span class="op">)</span><span class="co"># + theme(legend.position = "bottom")</span>
<span class="va">plts</span><span class="op">[[</span><span class="fl">4</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">simplot</span><span class="op">(</span><span class="va">fas</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span>, <span class="st">'CF_YL21'</span><span class="op">)</span>
<span class="fu">plot_list</span><span class="op">(</span>gglist<span class="op">=</span><span class="va">plts</span>, ncol<span class="op">=</span><span class="fl">1</span>, tag_levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"A"</span>,<span class="st">' '</span>,<span class="st">"B"</span>, <span class="st">' '</span>,<span class="st">"C"</span>, <span class="st">' '</span>, <span class="st">"D"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/unnamed-chunk-14-1.png" width="960"></p>
</details>
</div>
<div id="探索msa和相关数据的联系" class="section level2">
<h2 class="hasAnchor">
<a href="#%E6%8E%A2%E7%B4%A2msa%E5%92%8C%E7%9B%B8%E5%85%B3%E6%95%B0%E6%8D%AE%E7%9A%84%E8%81%94%E7%B3%BB" class="anchor"></a>探索MSA和相关数据的联系</h2>
<p>通过<code>ggtree</code>或者<code>ggtreeExtra</code>分面功能，形成Tree-MSA-External 的图形结构， 从可视化的角度观察MSA和特殊的表型之间的联系。 <img src="images/Fig2.png"></p>
<p>通过<code>ggtree(tree) + geom_facet(geom_msa) + geom_facet(ohters)</code>或者 <code>ggtree(tree) + geom_fruit(geom_msa) + geom_fruit(others)</code>的代码模式来整合外部 数据。</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#load MSA and genes locus</span>
<span class="va">tp53_sequences</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"tp53.fa"</span>, package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>
<span class="va">tp53_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TP53_genes.xlsx"</span>, package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>

<span class="co">#generating phylogenetic tree according to MSA</span>
<span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu">read.aa</span><span class="op">(</span><span class="va">tp53_sequences</span>, format <span class="op">=</span> <span class="st">"fasta"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">phyDat</span><span class="op">(</span>type <span class="op">=</span> <span class="st">"AA"</span>, levels <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu">dist.ml</span><span class="op">(</span><span class="va">dat</span>, model <span class="op">=</span> <span class="st">"JTT"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">bionj</span><span class="op">(</span><span class="op">)</span>
<span class="va">dd</span> <span class="op">&lt;-</span> <span class="fu">ggimage</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggimage/man/phylopic_uid.html">phylopic_uid</a></span><span class="op">(</span><span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span>

<span class="co">#visualizing phylogenetic tree by ggtree</span>
<span class="va">p_tp53</span> <span class="op">&lt;-</span> <span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span>, branch.length <span class="op">=</span> <span class="st">'none'</span><span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">dd</span> <span class="op">+</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>image<span class="op">=</span><span class="va">uid</span><span class="op">)</span>, geom <span class="op">=</span> <span class="st">"phylopic"</span>, offset <span class="op">=</span><span class="fl">1.9</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>label<span class="op">=</span><span class="va">label</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_treescale</span><span class="op">(</span>x <span class="op">=</span> <span class="fl">0</span>,y <span class="op">=</span> <span class="op">-</span><span class="fl">1</span><span class="op">)</span>

<span class="co">#deal with MSA data</span>
<span class="va">data_53</span> <span class="op">&lt;-</span> <span class="fu">readAAMultipleAlignment</span><span class="op">(</span><span class="va">tp53_sequences</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">tidy_msa</span><span class="op">(</span><span class="op">)</span>

<span class="co">#gene maps</span>
<span class="va">TP53_arrow</span> <span class="op">&lt;-</span> <span class="fu">read_xlsx</span><span class="op">(</span><span class="va">tp53_genes</span><span class="op">)</span>
<span class="va">TP53_arrow</span><span class="op">$</span><span class="va">direction</span> <span class="op">&lt;-</span> <span class="fl">1</span>
<span class="va">TP53_arrow</span><span class="op">[</span><span class="va">TP53_arrow</span><span class="op">$</span><span class="va">strand</span> <span class="op">==</span> <span class="st">"reverse"</span>,<span class="st">"direction"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="op">-</span><span class="fl">1</span>

<span class="co">#color</span>
<span class="va">mapping</span> <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, fill <span class="op">=</span> <span class="va">gene</span>, forward <span class="op">=</span> <span class="va">direction</span><span class="op">)</span>
<span class="va">my_pal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rev.html">rev</a></span><span class="op">(</span><span class="fu">brewer.pal</span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span>, name <span class="op">=</span> <span class="st">"Set3"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co">#tree + gene maps + msa</span>
<span class="va">p6a</span> <span class="op">&lt;-</span> <span class="va">p_tp53</span>  <span class="op">+</span> <span class="fu">xlim_tree</span><span class="op">(</span><span class="fl">4</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_facet</span><span class="op">(</span>geom <span class="op">=</span> <span class="va">geom_msa</span>, data <span class="op">=</span> <span class="va">data_53</span>,
             panel <span class="op">=</span> <span class="st">'Multiple Sequence Alignment of the TP53 Protein'</span>, font <span class="op">=</span> <span class="cn">NULL</span>,
             border <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">new_scale_fill</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_fill_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="fu">my_pal</span><span class="op">(</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_facet</span><span class="op">(</span>geom <span class="op">=</span> <span class="va">geom_motif</span>,
             mapping <span class="op">=</span> <span class="va">mapping</span>, data <span class="op">=</span> <span class="va">TP53_arrow</span>,
             panel <span class="op">=</span> <span class="st">'Genome_Locus'</span>,  on <span class="op">=</span> <span class="st">'TP53'</span>,
             arrowhead_height <span class="op">=</span> <span class="fu">unit</span><span class="op">(</span><span class="fl">3</span>, <span class="st">"mm"</span><span class="op">)</span>,
             arrowhead_width <span class="op">=</span> <span class="fu">unit</span><span class="op">(</span><span class="fl">1</span>, <span class="st">"mm"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme</span><span class="op">(</span>strip.background<span class="op">=</span><span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        strip.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">13</span><span class="op">)</span><span class="op">)</span>
<span class="va">p6A</span> <span class="op">&lt;-</span> <span class="fu">facet_widths</span><span class="op">(</span><span class="va">p6a</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>Tree <span class="op">=</span> <span class="fl">0.35</span>, Genome_Locus <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span><span class="op">)</span>
<span class="va">p6A</span></code></pre></div>
<details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/unnamed-chunk-15-1.png" width="1440"></p>
</details><p>可以结合<a href="https://www.ncbi.nlm.nih.gov/clinvar/">ClinVar Search database</a>探索更多的MSA与表型的关系。 <img src="images/ClinVar.png"></p>
</div>
</div>
<div id="总结" class="section level1">
<h1 class="hasAnchor">
<a href="#%E6%80%BB%E7%BB%93" class="anchor"></a>总结</h1>
<p>可视化探索MSA是分子生物学的重要内容，但在堆叠的MSA图形中表现并不佳。我们开发的R 包ggmsa，用于可视化地探索MSA和相关数据。它包含一系列通用MSA可视化方法，并通过将 多个相关数据集成到堆叠MSA图形中，扩展了堆叠MSA可视化方法的使用范围。堆叠的MSA图形 集成多个外部数据，在视觉层面探索序列特征的相关性。因此，将不同类型的外部数据与 MSA相结合，可以观察到不同的相关性。对同一个MSA数据采用多种可视化方法，可以从不同 角度探索序列特征。ggmsa中的sequence logos和sequece bundles能快速识别出MSA中的 sequence pattern和残基相关性。最后的碱基差异性和相似性图谱，用于鉴定序列重组事件。 因此，ggmsa代表了对以前拓展性受限的堆叠可视化方法的和可视化方法应用单一的工具的 实质性改进。(<a href="https://yulab-smu.top/" class="uri">https://yulab-smu.top/</a>)</p>
<p><img src="images/pipeline.png"></p>
<div id="session-information" class="section level2">
<h2 class="hasAnchor">
<a href="#session-information" class="anchor"></a>8. Session information</h2>
<p>以下是本次workshop中系统，R版本以及使用的包的信息：</p>
<details><summary>
output
</summary><pre><code>#&gt; R version 4.1.1 (2021-08-10)
#&gt; Platform: x86_64-pc-linux-gnu (64-bit)
#&gt; Running under: Ubuntu 20.04.3 LTS
#&gt; 
#&gt; Matrix products: default
#&gt; BLAS/LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.8.so
#&gt; 
#&gt; locale:
#&gt;  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#&gt;  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#&gt;  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=C             
#&gt;  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#&gt;  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
#&gt; [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
#&gt; 
#&gt; attached base packages:
#&gt; [1] stats4    stats     graphics  grDevices utils     datasets  methods  
#&gt; [8] base     
#&gt; 
#&gt; other attached packages:
#&gt;  [1] kableExtra_1.3.4    dplyr_1.0.7         RColorBrewer_1.1-2 
#&gt;  [4] ggnewscale_0.4.5    Biostrings_2.62.0   GenomeInfoDb_1.30.0
#&gt;  [7] XVector_0.34.0      IRanges_2.28.0      S4Vectors_0.32.0   
#&gt; [10] BiocGenerics_0.40.0 readxl_1.3.1        ggtree_3.2.0       
#&gt; [13] ggimage_0.3.0       phangorn_2.7.1      ape_5.5            
#&gt; [16] aplot_0.1.1         ggplot2_3.3.5       ggmsa_1.0.0        
#&gt; 
#&gt; loaded via a namespace (and not attached):
#&gt;  [1] nlme_3.1-153           bitops_1.0-7           fs_1.5.0              
#&gt;  [4] webshot_0.5.2          httr_1.4.2             ash_1.0-15            
#&gt;  [7] rprojroot_2.0.2        tools_4.1.1            utf8_1.2.2            
#&gt; [10] R6_2.5.1               KernSmooth_2.23-20     lazyeval_0.2.2        
#&gt; [13] colorspace_2.0-2       withr_2.4.2            tidyselect_1.1.1      
#&gt; [16] ggalt_0.4.0            curl_4.3.2             compiler_4.1.1        
#&gt; [19] extrafontdb_1.0        rvest_1.0.2            textshaping_0.3.6     
#&gt; [22] xml2_1.3.2             desc_1.4.0             labeling_0.4.2        
#&gt; [25] scales_1.1.1           proj4_1.0-10.1         quadprog_1.5-8        
#&gt; [28] pkgdown_1.6.1          systemfonts_1.0.3      stringr_1.4.0         
#&gt; [31] digest_0.6.28          yulab.utils_0.0.4      R4RNA_1.22.0          
#&gt; [34] svglite_2.0.0          rmarkdown_2.11         pkgconfig_2.0.3       
#&gt; [37] htmltools_0.5.2        extrafont_0.17         highr_0.9             
#&gt; [40] fastmap_1.1.0          maps_3.4.0             rlang_0.4.12          
#&gt; [43] rstudioapi_0.13        gridGraphics_0.5-1     jquerylib_0.1.4       
#&gt; [46] farver_2.1.0           generics_0.1.1         jsonlite_1.7.2        
#&gt; [49] RCurl_1.98-1.5         magrittr_2.0.1         ggplotify_0.1.0       
#&gt; [52] GenomeInfoDbData_1.2.7 patchwork_1.1.1        Matrix_1.3-4          
#&gt; [55] Rcpp_1.0.7             munsell_0.5.0          fansi_0.5.0           
#&gt; [58] ggfittext_0.9.1        lifecycle_1.0.1        stringi_1.7.5         
#&gt; [61] yaml_2.2.1             seqmagick_0.1.5        MASS_7.3-54           
#&gt; [64] zlibbioc_1.40.0        grid_4.1.1             parallel_4.1.1        
#&gt; [67] crayon_1.4.2           lattice_0.20-45        magick_2.7.3          
#&gt; [70] knitr_1.36             pillar_1.6.4           igraph_1.2.7          
#&gt; [73] codetools_0.2-18       fastmatch_1.1-3        glue_1.4.2            
#&gt; [76] evaluate_0.14          ggfun_0.0.4            vctrs_0.3.8           
#&gt; [79] treeio_1.18.0          tweenr_1.0.2           gggenes_0.4.1         
#&gt; [82] cellranger_1.1.0       Rttf2pt1_1.3.9         gtable_0.3.0          
#&gt; [85] purrr_0.3.4            polyclip_1.10-0        tidyr_1.1.4           
#&gt; [88] cachem_1.0.6           xfun_0.27              ggforce_0.3.3         
#&gt; [91] tidytree_0.3.5         viridisLite_0.4.0      ragg_1.2.0            
#&gt; [94] tibble_3.1.5           memoise_2.0.0          ellipsis_0.3.2</code></pre>
</details>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Lang Zhou, Guangchuang Yu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
