<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>使用ggmsa可视化和探索多序列比对和相关数据 • ggmsaABiocWorkshop2021</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://cdnjs.cloudflare.com/ajax/libs/bootswatch/3.4.0/flatly/bootstrap.min.css" rel="stylesheet" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css">
<script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="使用ggmsa可视化和探索多序列比对和相关数据">
<meta property="og:description" content="ggmsaABiocWorkshop2021">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    <div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">ggmsaABiocWorkshop2021</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="Released version">0.9.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fas fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/HOWTO_BUILD_WORKSHOP.html">How To Build A Workshop Package</a>
    </li>
    <li>
      <a href="../articles/workflow_ggmsa_Chinese.html">使用ggmsa可视化和探索多序列比对和相关数据</a>
    </li>
    <li>
      <a href="../articles/workshop_example.html">An Example Workshop</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right">
<li>
  <a href="https://github.com/nyzhoulang/ggmsaBioWorkshop2021">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><script src="workflow_ggmsa_Chinese_files/header-attrs-2.11/header-attrs.js"></script><script src="workflow_ggmsa_Chinese_files/kePrint-0.0.1/kePrint.js"></script><link href="workflow_ggmsa_Chinese_files/lightable-0.0.1/lightable.css" rel="stylesheet">
<div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1 data-toc-skip>使用ggmsa可视化和探索多序列比对和相关数据</h1>
                        <h4 class="author">Lang Chau, Guangchuang Yu, Shuangbin Xu</h4>
            
            <h4 class="date">Oct 2021</h4>
      
      <small class="dont-index">Source: <a href="https://github.com/nyzhoulang/ggmsaBioWorkshop2021/blob/master/vignettes/workflow_ggmsa_Chinese.Rmd"><code>vignettes/workflow_ggmsa_Chinese.Rmd</code></a></small>
      <div class="hidden name"><code>workflow_ggmsa_Chinese.Rmd</code></div>

    </div>

    
    
<div id="背景介绍" class="section level1">
<h1 class="hasAnchor">
<a href="#%E8%83%8C%E6%99%AF%E4%BB%8B%E7%BB%8D" class="anchor"></a>背景介绍</h1>
<div id="摘要" class="section level2">
<h2 class="hasAnchor">
<a href="#%E6%91%98%E8%A6%81" class="anchor"></a>摘要</h2>
<p>在多序列比对(Multiple Sequence Alignment,MSA)的可视化中，普遍采用堆叠可视化的形 式，其以行表示每条序列，以列表示相同位置的不同残基字符。我们开发的R包ggmsa在 可视化层面扩展了这种堆叠形式的可视化方法，使得MSA图形可以与各类数据进行整合。 例如：允许将序列二级结构信息、基因座位点信息或表型数据等序列相关的数据集与MSA结 合，以此来探索序列-结构-功能三者之间的关系。并且，我们还开发和整合多种特别的 比对序列可视化方法，包核苷酸差异图、核苷酸相似性图 (鉴定序列重组信号)、 序列集束(Sequence Bundles)和序列标识(Sequence Logos)， 这允许用户从不同角度探索序列的特征。</p>
</div>
<div id="生物学问题描述" class="section level2">
<h2 class="hasAnchor">
<a href="#%E7%94%9F%E7%89%A9%E5%AD%A6%E9%97%AE%E9%A2%98%E6%8F%8F%E8%BF%B0" class="anchor"></a>生物学问题描述</h2>
<div id="issues-1" class="section level3">
<h3 class="hasAnchor">
<a href="#issues-1" class="anchor"></a>issues 1</h3>
<p>使用单一的可视化范式很难捕获隐藏在排列中的令人满意的分子特征。</p>
<details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/unnamed-chunk-2-1.png" width="3300"></p>
</details><p>堆叠图形在展示局部MSA时有很大优势，它以精确到单碱基/氨基酸水平的分辨率展示比对， 并通过染色让用户能快速识别比对的变化趋势。</p>
<details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/unnamed-chunk-3-1.png" width="3000"></p>
</details><p>但是当比对序列比较大比较复杂时，会让人产生视觉上的混乱感。因为图形会包含密集的 字符和令人眼花缭乱的配色。</p>
</div>
<div id="issues-2" class="section level3">
<h3 class="hasAnchor">
<a href="#issues-2" class="anchor"></a>issues 2</h3>
<p>现有工具缺乏同时探索MSA和相关数据的能力。</p>
<p><img src="images/issus2.png#pic_center"></p>
<p>我们在探索MSA的序列特征时，往往试图深入序列的系统发育关系和结构功能关系。 因为它们时相互联系，相互影响。但是这在可视化层面就无法完成, 我们没有办法在绘图时把序列和其他数据联系起来。</p>
</div>
</div>
</div>
<div id="使用ggmsa探索msa和相关数据" class="section level1">
<h1 class="hasAnchor">
<a href="#%E4%BD%BF%E7%94%A8ggmsa%E6%8E%A2%E7%B4%A2msa%E5%92%8C%E7%9B%B8%E5%85%B3%E6%95%B0%E6%8D%AE" class="anchor"></a>使用ggmsa探索MSA和相关数据</h1>
<div id="ggmsa的技术路线" class="section level2">
<h2 class="hasAnchor">
<a href="#ggmsa%E7%9A%84%E6%8A%80%E6%9C%AF%E8%B7%AF%E7%BA%BF" class="anchor"></a>ggmsa的技术路线</h2>
<p><img src="images/pipeline.png#pic_center"></p>
<p>ggmsa通过两种方案来解决上面的问题。 首先，单一的可视化方法无法得到令人满意的可视化方法，因此ggmsa开发并整合多种 通用的序列可视化方法。这为用户提供了更多可视化手段来从不同角度探索序列数据。</p>
<p>其次，ggmsa 拓展了普通的堆叠图形，使得外部数据可以对齐到MSA的行或者列上，实现 在可视化层面同时探索序列和相关数据。</p>
</div>
<div id="不同的可视化方法" class="section level2">
<h2 class="hasAnchor">
<a href="#%E4%B8%8D%E5%90%8C%E7%9A%84%E5%8F%AF%E8%A7%86%E5%8C%96%E6%96%B9%E6%B3%95" class="anchor"></a>不同的可视化方法</h2>
<div id="堆叠图形" class="section level3">
<h3 class="hasAnchor">
<a href="#%E5%A0%86%E5%8F%A0%E5%9B%BE%E5%BD%A2" class="anchor"></a>堆叠图形</h3>
<details><summary>
plot
</summary><p><img src="images/stacked_MSA.png" width="30%" style="display: block; margin: auto;"></p>
</details><div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">protein_sequences</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sample.fasta"</span>, package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>
<span class="fu">readAAMultipleAlignment</span><span class="op">(</span><span class="va">protein_sequences</span><span class="op">)</span> </code></pre></div>
<details><summary>
output
</summary><pre><code>#&gt; AAMultipleAlignment with 9 rows and 456 columns
#&gt;      aln                                                    names               
#&gt; [1] MAAVVLENGVLSRKLSDFGQETSYIE...QLKILADSINSEVGILCNALQKIKS PH4H_Rattus_norve...
#&gt; [2] MAAVVLENGVLSRKLSDFGQETSYIE...QLKILADSINSEVGILCHALQKIKS PH4H_Mus_musculus
#&gt; [3] MSTAVLENPGLGRKLSDFGQETSYIE...QLKILADSINSEIGILCSALQKIK- PH4H_Homo_sapiens
#&gt; [4] MSALVLESRALGRKLSDFGQETSYIE...QLKILADSISSEVEILCSALQKLK- PH4H_Bos_taurus
#&gt; [5] --------------------------...LNAGDRQGWADTEDV---------- PH4H_Chromobacter...
#&gt; [6] --------------------------...LNAGTREGWADTADI---------- PH4H_Ralstonia_so...
#&gt; [7] --------------------------...LTRGT-QAYATAGGRLAGAAAG--- PH4H_Caulobacter_...
#&gt; [8] --------------------------...------------------------- PH4H_Pseudomonas_...
#&gt; [9] --------------------------...------------------------- PH4H_Rhizobium_loti</code></pre>
</details><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">ggmsa</span><span class="op">(</span><span class="va">protein_sequences</span>, 
      start <span class="op">=</span> <span class="fl">221</span>, 
      end <span class="op">=</span> <span class="fl">280</span>, 
      char_width <span class="op">=</span> <span class="fl">0.5</span>,
      seq_name <span class="op">=</span> <span class="cn">TRUE</span>,
      border <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span></code></pre></div>
<details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/unnamed-chunk-5-1.png" width="3300"></p>
</details><p>这是最简单的序列可视化方法 应用的颜色，字体，输入数据类型</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="fu">available_colors</span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; 1.color schemes for nucleotide sequences currently available:</span>
<span class="co">#&gt; Chemistry_NT Shapely_NT Taylor_NT Zappo_NT</span>
<span class="co">#&gt; 2.color schemes for AA sequences currently available:</span>
<span class="co">#&gt; ClustalChemistry_AA Shapely_AA Zappo_AA Taylor_AA LETTER CN6 Hydrophobicity</span>
<span class="fu">available_fonts</span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; font families currently available:</span>
<span class="co">#&gt; helvetical mono TimesNewRoman DroidSansMono</span>
<span class="fu">available_msa</span><span class="op">(</span><span class="op">)</span>
<span class="co">#&gt; 1.files currently available:</span>
<span class="co">#&gt; .fasta</span>
<span class="co">#&gt; 2.XStringSet objects from 'Biostrings' package:</span>
<span class="co">#&gt; DNAStringSet RNAStringSet AAStringSet BStringSet DNAMultipleAlignment RNAMultipleAlignment AAMultipleAlignment</span>
<span class="co">#&gt; 3.bin objects:</span>
<span class="co">#&gt; DNAbin AAbin</span></code></pre></div>
</div>
<div id="堆叠图形中的注释模块" class="section level3">
<h3 class="hasAnchor">
<a href="#%E5%A0%86%E5%8F%A0%E5%9B%BE%E5%BD%A2%E4%B8%AD%E7%9A%84%E6%B3%A8%E9%87%8A%E6%A8%A1%E5%9D%97" class="anchor"></a>堆叠图形中的注释模块</h3>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">protein_sequences</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"sample.fasta"</span>, package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>
<span class="fu">ggmsa</span><span class="op">(</span><span class="va">protein_sequences</span>, start <span class="op">=</span> <span class="fl">221</span>, end <span class="op">=</span> <span class="fl">280</span>, char_width <span class="op">=</span> <span class="fl">0.5</span>, seq_name <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_seqlogo</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span> <span class="fu">geom_msaBar</span><span class="op">(</span><span class="op">)</span></code></pre></div>
<details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/unnamed-chunk-7-1.png" width="3300"></p>
</details><table class="table table">
<thead><tr>
<th style="text-align:left;">
Annotation modules
</th>
<th style="text-align:left;">
Type
</th>
<th style="text-align:left;">
Description
</th>
</tr></thead>
<tbody>
<tr>
<td style="text-align:left;">
geom_seqlogo()
</td>
<td style="text-align:left;">
geometric layer
</td>
<td style="text-align:left;">
automatically generated sequence logos for a MSA
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_GC()
</td>
<td style="text-align:left;">
annotation module
</td>
<td style="text-align:left;">
shows GC content with bubble chart
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_seed()
</td>
<td style="text-align:left;">
annotation module
</td>
<td style="text-align:left;">
highlights seed region on miRNA sequences
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_msaBar()
</td>
<td style="text-align:left;">
annotation module
</td>
<td style="text-align:left;">
shows sequences conservation by a bar chart
</td>
</tr>
<tr>
<td style="text-align:left;">
geom_helix()
</td>
<td style="text-align:left;">
annotation module
</td>
<td style="text-align:left;">
depicts RNA secondary structure as arc diagrams(need extra data)
</td>
</tr>
<tr>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
<td style="text-align:left;">
</td>
</tr>
</tbody>
</table>
<p>不同的布局</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="co"># 4 fields</span>
 <span class="fu">ggmsa</span><span class="op">(</span><span class="va">protein_sequences</span>, start <span class="op">=</span> <span class="fl">0</span>, end <span class="op">=</span> <span class="fl">400</span>, font <span class="op">=</span> <span class="cn">NULL</span>, color <span class="op">=</span> <span class="st">"Chemistry_AA"</span><span class="op">)</span> <span class="op">+</span> 
    <span class="fu">facet_msa</span><span class="op">(</span>field <span class="op">=</span> <span class="fl">100</span><span class="op">)</span></code></pre></div>
<details><summary>
plot
</summary><div class="figure">
<img src="workflow_ggmsa_Chinese_files/figure-html/facet-1.png" alt="\label{fig:facet} The long sequence was broken down and displayed in several lines." width="960"><p class="caption">
 The long sequence was broken down and displayed in several lines.
</p>
</div>
</details>
</div>
<div id="sequence-bundle" class="section level3">
<h3 class="hasAnchor">
<a href="#sequence-bundle" class="anchor"></a>Sequence bundle</h3>
<details><summary>
plot
</summary><p><img src="images/seq_bundle.png" width="30%" style="display: block; margin: auto;"></p>
</details><details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/unnamed-chunk-10-1.png" width="1056"></p>
</details>
</div>
<div id="sequence-logo" class="section level3">
<h3 class="hasAnchor">
<a href="#sequence-logo" class="anchor"></a>Sequence logo</h3>
<details><summary>
plot
</summary><p><img src="images/seq_logo.png" width="30%" style="display: block; margin: auto;"></p>
</details><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="fu">ggmsa</span><span class="op">(</span><span class="va">protein_sequences</span>, <span class="fl">221</span>, <span class="fl">280</span>, seq_name <span class="op">=</span> <span class="cn">TRUE</span>, char_width <span class="op">=</span> <span class="fl">0.5</span>, font <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span> <span class="op">+</span>
    <span class="fu">geom_seqlogo</span><span class="op">(</span>color <span class="op">=</span> <span class="st">"Chemistry_AA"</span><span class="op">)</span> </code></pre></div>
<details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/unnamed-chunk-12-1.png" width="1056"></p>
</details><div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"> <span class="fu">seqlogo</span><span class="op">(</span><span class="va">protein_sequences</span>, <span class="fl">221</span>, <span class="fl">280</span><span class="op">)</span></code></pre></div>
<details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/unnamed-chunk-13-1.png" width="768"></p>
</details>
</div>
<div id="sequence-recombination" class="section level3">
<h3 class="hasAnchor">
<a href="#sequence-recombination" class="anchor"></a>Sequence recombination</h3>
<details><summary>
plot
</summary><p><img src="images/recombination.png" width="30%" height="50%" style="display: block; margin: auto;"></p>
</details><div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">fas</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.files.html">list.files</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"GVariation"</span>, package<span class="op">=</span><span class="st">"ggmsa"</span><span class="op">)</span>,
                  pattern<span class="op">=</span><span class="st">"fas"</span>, full.names<span class="op">=</span><span class="cn">TRUE</span><span class="op">)</span>
<span class="va">xx</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">fas</span>, <span class="va">seqdiff</span><span class="op">)</span>
<span class="va">plts</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/lapply.html">lapply</a></span><span class="op">(</span><span class="va">xx</span>, <span class="va">plot</span>, width <span class="op">=</span> <span class="fl">100</span><span class="op">)</span>
<span class="va">fas</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata/GVariation/sample_alignment.fa"</span>, 
                    package<span class="op">=</span><span class="st">"ggmsa"</span><span class="op">)</span><span class="co"># + theme(legend.position = "bottom")</span>
<span class="va">plts</span><span class="op">[[</span><span class="fl">4</span><span class="op">]</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="fu">simplot</span><span class="op">(</span><span class="va">fas</span><span class="op">[</span><span class="fl">4</span><span class="op">]</span>, <span class="st">'CF_YL21'</span><span class="op">)</span>
<span class="fu">plot_list</span><span class="op">(</span>gglist<span class="op">=</span><span class="va">plts</span>, ncol<span class="op">=</span><span class="fl">1</span>, tag_levels <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="st">"A"</span>,<span class="st">' '</span>,<span class="st">"B"</span>, <span class="st">' '</span>,<span class="st">"C"</span>, <span class="st">' '</span>, <span class="st">"D"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span></code></pre></div>
<details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/unnamed-chunk-15-1.png" width="960"></p>
</details>
</div>
</div>
<div id="拓展stacked-msa" class="section level2">
<h2 class="hasAnchor">
<a href="#%E6%8B%93%E5%B1%95stacked-msa" class="anchor"></a>拓展Stacked MSA</h2>
<p><img src="images/Fig2.png"></p>
<div id="对齐到行" class="section level3">
<h3 class="hasAnchor">
<a href="#%E5%AF%B9%E9%BD%90%E5%88%B0%E8%A1%8C" class="anchor"></a>对齐到行</h3>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="co">#load MSA and genes locus</span>
<span class="va">tp53_sequences</span> <span class="op">&lt;-</span>  <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"tp53.fa"</span>, package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>
<span class="va">tp53_genes</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"TP53_genes.xlsx"</span>, package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>

<span class="co">#generating phylogenetic tree according to MSA</span>
<span class="va">dat</span> <span class="op">&lt;-</span> <span class="fu">read.aa</span><span class="op">(</span><span class="va">tp53_sequences</span>, format <span class="op">=</span> <span class="st">"fasta"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">phyDat</span><span class="op">(</span>type <span class="op">=</span> <span class="st">"AA"</span>, levels <span class="op">=</span> <span class="cn">NULL</span><span class="op">)</span>
<span class="va">tree</span> <span class="op">&lt;-</span> <span class="fu">dist.ml</span><span class="op">(</span><span class="va">dat</span>, model <span class="op">=</span> <span class="st">"JTT"</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">bionj</span><span class="op">(</span><span class="op">)</span>
<span class="va">dd</span> <span class="op">&lt;-</span> <span class="fu">ggimage</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/ggimage/man/phylopic_uid.html">phylopic_uid</a></span><span class="op">(</span><span class="va">tree</span><span class="op">$</span><span class="va">tip.label</span><span class="op">)</span>

<span class="co">#visualizing phylogenetic tree by ggtree</span>
<span class="va">p_tp53</span> <span class="op">&lt;-</span> <span class="fu">ggtree</span><span class="op">(</span><span class="va">tree</span>, branch.length <span class="op">=</span> <span class="st">'none'</span><span class="op">)</span> <span class="op">%&lt;+%</span> <span class="va">dd</span> <span class="op">+</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>image<span class="op">=</span><span class="va">uid</span><span class="op">)</span>, geom <span class="op">=</span> <span class="st">"phylopic"</span>, offset <span class="op">=</span><span class="fl">1.9</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_tiplab</span><span class="op">(</span><span class="fu">aes</span><span class="op">(</span>label<span class="op">=</span><span class="va">label</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_treescale</span><span class="op">(</span>x <span class="op">=</span> <span class="fl">0</span>,y <span class="op">=</span> <span class="op">-</span><span class="fl">1</span><span class="op">)</span>

<span class="co">#deal with MSA data</span>
<span class="va">data_53</span> <span class="op">&lt;-</span> <span class="fu">readAAMultipleAlignment</span><span class="op">(</span><span class="va">tp53_sequences</span><span class="op">)</span> <span class="op">%&gt;%</span> <span class="fu">tidy_msa</span><span class="op">(</span><span class="op">)</span>

<span class="co">#gene maps</span>
<span class="va">TP53_arrow</span> <span class="op">&lt;-</span> <span class="fu">read_xlsx</span><span class="op">(</span><span class="va">tp53_genes</span><span class="op">)</span>
<span class="va">TP53_arrow</span><span class="op">$</span><span class="va">direction</span> <span class="op">&lt;-</span> <span class="fl">1</span>
<span class="va">TP53_arrow</span><span class="op">[</span><span class="va">TP53_arrow</span><span class="op">$</span><span class="va">strand</span> <span class="op">==</span> <span class="st">"reverse"</span>,<span class="st">"direction"</span><span class="op">]</span> <span class="op">&lt;-</span> <span class="op">-</span><span class="fl">1</span>

<span class="co">#color</span>
<span class="va">mapping</span> <span class="op">=</span> <span class="fu">aes</span><span class="op">(</span>xmin <span class="op">=</span> <span class="va">start</span>, xmax <span class="op">=</span> <span class="va">end</span>, fill <span class="op">=</span> <span class="va">gene</span>, forward <span class="op">=</span> <span class="va">direction</span><span class="op">)</span>
<span class="va">my_pal</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/grDevices/colorRamp.html">colorRampPalette</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/rev.html">rev</a></span><span class="op">(</span><span class="fu">brewer.pal</span><span class="op">(</span>n <span class="op">=</span> <span class="fl">10</span>, name <span class="op">=</span> <span class="st">"Set3"</span><span class="op">)</span><span class="op">)</span><span class="op">)</span>

<span class="co">#tree + gene maps + msa</span>
<span class="va">p6a</span> <span class="op">&lt;-</span> <span class="va">p_tp53</span>  <span class="op">+</span> <span class="fu">xlim_tree</span><span class="op">(</span><span class="fl">4</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_facet</span><span class="op">(</span>geom <span class="op">=</span> <span class="va">geom_msa</span>, data <span class="op">=</span> <span class="va">data_53</span>,
             panel <span class="op">=</span> <span class="st">'Multiple Sequence Alignment of the TP53 Protein'</span>, font <span class="op">=</span> <span class="cn">NULL</span>,
             border <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">new_scale_fill</span><span class="op">(</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">scale_fill_manual</span><span class="op">(</span>values <span class="op">=</span> <span class="fu">my_pal</span><span class="op">(</span><span class="fl">10</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_facet</span><span class="op">(</span>geom <span class="op">=</span> <span class="va">geom_motif</span>,
             mapping <span class="op">=</span> <span class="va">mapping</span>, data <span class="op">=</span> <span class="va">TP53_arrow</span>,
             panel <span class="op">=</span> <span class="st">'Genome_Locus'</span>,  on <span class="op">=</span> <span class="st">'TP53'</span>,
             arrowhead_height <span class="op">=</span> <span class="fu">unit</span><span class="op">(</span><span class="fl">3</span>, <span class="st">"mm"</span><span class="op">)</span>,
             arrowhead_width <span class="op">=</span> <span class="fu">unit</span><span class="op">(</span><span class="fl">1</span>, <span class="st">"mm"</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">theme</span><span class="op">(</span>strip.background<span class="op">=</span><span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span>,
        strip.text <span class="op">=</span> <span class="fu">element_text</span><span class="op">(</span>size <span class="op">=</span> <span class="fl">13</span><span class="op">)</span><span class="op">)</span>
<span class="va">p6A</span> <span class="op">&lt;-</span> <span class="fu">facet_widths</span><span class="op">(</span><span class="va">p6a</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span>Tree <span class="op">=</span> <span class="fl">0.35</span>, Genome_Locus <span class="op">=</span> <span class="fl">0.3</span><span class="op">)</span><span class="op">)</span>
<span class="va">p6A</span></code></pre></div>
<details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/unnamed-chunk-16-1.png" width="1440"></p>
</details>
</div>
<div id="对齐到列" class="section level3">
<h3 class="hasAnchor">
<a href="#%E5%AF%B9%E9%BD%90%E5%88%B0%E5%88%97" class="anchor"></a>对齐到列</h3>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span class="va">RNA7S</span>  <span class="op">&lt;-</span> <span class="st">"data/3JAJ-2D-dotbracket.txt"</span>
<span class="va">RNAP54</span> <span class="op">&lt;-</span> <span class="st">"data/4UJE-2D-dotbracket.txt"</span>

<span class="va">RF03120_msa</span><span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"Rfam"</span>, <span class="st">"RF03120.fasta"</span>, package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>
<span class="va">RF03120_ss</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"Rfam"</span>, <span class="st">"RF03120_SS.txt"</span>, package <span class="op">=</span> <span class="st">"ggmsa"</span><span class="op">)</span>

<span class="va">known</span> <span class="op">&lt;-</span> <span class="fu">readSSfile</span><span class="op">(</span><span class="va">RNA7S</span>, type <span class="op">=</span> <span class="st">"Vienna"</span> <span class="op">)</span>
<span class="va">transat</span> <span class="op">&lt;-</span> <span class="fu">readSSfile</span><span class="op">(</span><span class="va">RNAP54</span> , type <span class="op">=</span> <span class="st">"Vienna"</span><span class="op">)</span>


<span class="va">RF_arc</span> <span class="op">&lt;-</span> <span class="fu">readSSfile</span><span class="op">(</span><span class="va">RF03120_ss</span>, type <span class="op">=</span> <span class="st">"Vienna"</span> <span class="op">)</span>

<span class="va">p7A</span> <span class="op">&lt;-</span> <span class="fu">ggmsa</span><span class="op">(</span><span class="va">RF03120_msa</span>, 
             font <span class="op">=</span> <span class="cn">NULL</span>, 
             color <span class="op">=</span> <span class="st">"Chemistry_NT"</span>, 
             seq_name <span class="op">=</span> <span class="cn">F</span>, 
             show.legend <span class="op">=</span> <span class="cn">F</span>, 
             border <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op">+</span>
        <span class="fu">geom_helix</span><span class="op">(</span>helix_data <span class="op">=</span> <span class="va">RF_arc</span><span class="op">)</span> <span class="op">+</span> 
        <span class="fu">theme</span><span class="op">(</span>axis.text.y <span class="op">=</span> <span class="fu">element_blank</span><span class="op">(</span><span class="op">)</span><span class="op">)</span>

<span class="va">p7B</span> <span class="op">&lt;-</span> <span class="fu">ggmsa</span><span class="op">(</span><span class="st">"data/5SRNA.fa"</span>,
             font <span class="op">=</span> <span class="cn">NULL</span>,
             color <span class="op">=</span> <span class="st">"Chemistry_NT"</span>,
             seq_name <span class="op">=</span> <span class="cn">T</span>,
             show.legend <span class="op">=</span> <span class="cn">T</span>,
             border <span class="op">=</span> <span class="cn">NA</span><span class="op">)</span> <span class="op">+</span>
  <span class="fu">geom_helix</span><span class="op">(</span>helix_data <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span>known <span class="op">=</span> <span class="va">known</span>, 
                               predicted <span class="op">=</span> <span class="va">transat</span><span class="op">)</span>,
             overlap <span class="op">=</span> <span class="cn">F</span><span class="op">)</span>

<span class="va">p7</span> <span class="op">&lt;-</span> <span class="fu">plot_list</span><span class="op">(</span>gglist <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html">list</a></span><span class="op">(</span><span class="va">p7A</span>, <span class="va">p7B</span><span class="op">)</span>, 
                ncol <span class="op">=</span> <span class="fl">1</span>,
                heights <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html">c</a></span><span class="op">(</span><span class="fl">0.15</span><span class="op">)</span>, 
                tag_levels <span class="op">=</span> <span class="st">'A'</span><span class="op">)</span> 
<span class="va">p7</span></code></pre></div>
<details><summary>
plot
</summary><p><img src="workflow_ggmsa_Chinese_files/figure-html/unnamed-chunk-17-1.png" width="960"></p>
</details>
</div>
</div>
</div>
<div id="总结" class="section level1">
<h1 class="hasAnchor">
<a href="#%E6%80%BB%E7%BB%93" class="anchor"></a>总结</h1>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">

        <nav id="toc" data-toggle="toc"><h2 data-toc-skip>Contents</h2>
    </nav>
</div>

</div>



      <footer><div class="copyright">
  <p>Developed by Lang Zhou, Guangchuang Yu.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="https://pkgdown.r-lib.org/">pkgdown</a> 1.6.1.</p>
</div>

      </footer>
</div>

  


  </body>
</html>
